front/src/Connexion/LocalUserStore.ts

```diff
@@ -160,19 +164,24 @@ class LocalUserStore {
 
     verifyState(value: string): boolean {
         const oldValue = localStorage.getItem(state);
-        localStorage.removeItem(state);
         return oldValue === value;
     }
+    getState(): string | null {
+        return localStorage.getItem(state);
+    }
     generateNonce(): string {
         const newNonce = uuidv4();
         localStorage.setItem(nonce, newNonce);
         return newNonce;
     }
-
     getNonce(): string | null {
-        const oldValue = localStorage.getItem(nonce);
-        localStorage.removeItem(nonce);
-        return oldValue;
+        return localStorage.getItem(nonce);
+    }
+    setCode(value: string): void {
+        localStorage.setItem(code, value);
+    }
+    getCode(): string | null {
```

Comment:
the auth code should not be saved. It is supposed to be a temporary ressource used to get tokens

Reply:
Yes, I know it will be change